mods/map/init.lua

```diff
@@ -0,0 +1,66 @@
+map = {}
+
+
+-- Global function to allow overriding
+
+function map.check_for_map(player)
+	if creative.is_enabled_for(player:get_player_name()) or
+			player:get_inventory():contains_item("main", "map:map") then
+		player:hud_set_flags({minimap = true})
+	else
+		player:hud_set_flags({minimap = false})
+	end
+end
+
+
+-- Check for map on joinplayer
+
+minetest.register_on_joinplayer(function(player)
+	map.check_for_map(player)
+end)
+
+
+-- Regular check for map
+
+local timer = 0
+
+minetest.register_globalstep(function(dtime)
+	timer = timer + dtime
+	if timer < 7 then
+		return
+	end
+
+	timer = 0
+	for _, player in ipairs(minetest.get_connected_players()) do
```

Comment:
It's a common idiom to use minetest.after to run something at an interval

Reply:
Done.